app.rb

```diff
@@ -0,0 +1,173 @@
+require_relative 'person'
+require_relative 'teacher'
+require_relative 'student'
+require_relative 'book'
+require_relative 'rentable'
+
+class App
+  attr_accessor :persons, :books, :rentals
+
+  def initialize
+    @persons = []
+    @books = []
+    @rentals = []
+  end
+
+  def line_return
+    puts '---------------------------'
+  end
+
+  def print_books
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def print_books_by_index
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_books
+    @books.length.positive? ? print_books : (puts 'No books available')
+    line_return
+  end
+
+  def print_persons_by_index
+    @persons.each_with_index do |person, index|
+      puts "#{index}) [#{person.type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def print_persons
+    @persons.each do |person|
+      puts "[#{person.type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def list_people
+    @persons.length.positive? ? print_persons : (puts 'No person available')
+    line_return
+  end
+
+  def add_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp.capitalize
+    print 'Classroom: '
+    classroom = gets.chomp.capitalize
+    print 'Has parent permission? (Y/N): '
+    parent_permission = gets.chomp.downcase
+    while parent_permission != 'y' && parent_permission != 'n'
+      print 'Please input Y or N: '
+      parent_permission = gets.chomp.downcase
+    end
+    student = Student.new(age, name, classroom) if parent_permission == 'y'
+    student = Student.new(age, name, classroom, parent_permission: false) if parent_permission == 'n'
+    @persons << student
+    puts 'Student successfully created'
+    line_return
+  end
```

Comment:
- [ ] There is an error raised when I try to add a new student, Please fix that.

![image](https://github.com/codeWithHyder/my-school-library/assets/68796232/b76189a2-8cc6-41f3-b05c-7ec0d654bece)